### PR TITLE
Added regression test for Pojos with collections containing nulls

### PR DESCRIPTION
**Pull request by rozza**
JAVA-2756

### CODE REVIEW

File: bson/src/main/org/bson/codecs/pojo/CollectionPropertyCodecProvider.java [hunk 1]
```diff
@@ -63,7 +67,12 @@ public void encode(final BsonWriter writer, final Collection<T> collection, fina
             Collection<T> collection = getInstance();
             reader.readStartArray();
             while (reader.readBsonType() != BsonType.END_OF_DOCUMENT) {
-                collection.add(codec.decode(reader, context));
+                if (reader.getCurrentBsonType() == BsonType.NULL) {
+                    collection.add(null);
+                    reader.skipValue();
```
Comment (jyemin) [inline bson/src/main/org/bson/codecs/pojo/CollectionPropertyCodecProvider.java]: `reader.readNull()` makes the intent clearer than `reader.skipValue()`

File: bson/src/main/org/bson/codecs/pojo/MapPropertyCodecProvider.java [hunk 2]
```diff
@@ -69,7 +73,12 @@ public void encode(final BsonWriter writer, final Map<String, T> map, final Enco
             reader.readStartDocument();
             Map<String, T> map = getInstance();
             while (reader.readBsonType() != BsonType.END_OF_DOCUMENT) {
-                map.put(reader.readName(), codec.decode(reader, context));
+                if (reader.getCurrentBsonType() == BsonType.NULL) {
+                    map.put(reader.readName(), null);
+                    reader.skipValue();
```
Comment (jyemin) [inline bson/src/main/org/bson/codecs/pojo/MapPropertyCodecProvider.java]: Same comment as above.

**Comment by jyemin** [comment]
Oh, also please edit the commit description to make it clear that it's fixing a bug, not just adding a regression test for it.

**Comment by rozza** [comment]
Updated all, I've kept the original PR as well and merged.

**Pull request closed without merging**